src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -863,6 +868,14 @@ public function setSession(SessionInterface $session)
         $this->session = $session;
     }
 
+    /**
+     * @internal
+     */
+    public function setSessionFactory(callable $factory)
```

Comment:
This provides extra laziness, required by https://github.com/symfony/recipes/pull/333